lib/internal/abort_controller.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+// Modeled very closely on the AbortController implementation
+// in https://github.com/mysticatea/abort-controller (MIT license)
+
+const {
+  SafeWeakMap,
+  Object,
+  Symbol,
+} = primordials;
+const EventEmitter = require('events');
+const {
+  codes: { ERR_INVALID_THIS }
+} = require('internal/errors');
+
+const signals = new SafeWeakMap();
```

Comment:
There was a comment on #33556 that it might be better to use private properties or symbols, if that's possible then I'd do so here as well.

Reply:
Yep, that's the intent but I want to benchmark a few things first.